app/src/main/java/com/keylesspalace/tusky/components/account/list/ListSelectionFragment.kt

```diff
@@ -16,90 +16,116 @@
+import com.keylesspalace.tusky.databinding.FragmentListsListBinding
+import com.keylesspalace.tusky.databinding.ItemListBinding
 import com.keylesspalace.tusky.di.Injectable
 import com.keylesspalace.tusky.di.ViewModelFactory
+import com.keylesspalace.tusky.entity.MastoList
 import com.keylesspalace.tusky.util.BindingHolder
 import com.keylesspalace.tusky.util.hide
 import com.keylesspalace.tusky.util.show
-import com.keylesspalace.tusky.util.viewBinding
 import com.keylesspalace.tusky.util.visible
+import kotlinx.coroutines.CoroutineStart
+import kotlinx.coroutines.awaitCancellation
+import kotlinx.coroutines.delay
 import kotlinx.coroutines.flow.collectLatest
 import kotlinx.coroutines.launch
 import java.io.IOException
 import javax.inject.Inject
 
-class ListsForAccountFragment : DialogFragment(), Injectable {
+class ListSelectionFragment : DialogFragment(), Injectable {
+
+    interface ListSelectionListener {
+        fun onListSelected(list: MastoList)
+    }
 
     @Inject
     lateinit var viewModelFactory: ViewModelFactory
 
     private val viewModel: ListsForAccountViewModel by viewModels { viewModelFactory }
-    private val binding by viewBinding(FragmentListsForAccountBinding::bind)
+
+    private var _binding: FragmentListsListBinding? = null
+
+    // This property is only valid between onCreateDialog and onDestroyView
+    private val binding get() = _binding!!
 
     private val adapter = Adapter()
 
+    private var selectListener: ListSelectionListener? =  null
+    private var accountId: String? = null
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        selectListener = context as? ListSelectionListener
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setStyle(STYLE_NORMAL, R.style.TuskyDialogFragmentStyle)
-
-        viewModel.setup(requireArguments().getString(ARG_ACCOUNT_ID)!!)
+        accountId = requireArguments().getString(ARG_ACCOUNT_ID)
     }
 
-    override fun onStart() {
-        super.onStart()
-        dialog?.apply {
-            window?.setLayout(
-                LinearLayout.LayoutParams.MATCH_PARENT,
-                LinearLayout.LayoutParams.MATCH_PARENT
-            )
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val context = requireContext()
+
+        _binding = FragmentListsListBinding.inflate(layoutInflater)
+        binding.listsView.adapter = adapter
+
+        val dialogBuilder = AlertDialog.Builder(context)
+            .setView(binding.root)
+            .setTitle(R.string.select_list_title)
+            .setNeutralButton(R.string.select_list_manage) { _, _ ->
+                val listIntent = Intent(context, ListsActivity::class.java)
+                startActivity(listIntent)
+            }
+            .setNegativeButton(if (accountId != null) R.string.button_done else android.R.string.cancel, null)
```

Comment:
Generally this fragment now serves two leaders (simple list selection or list member management). This is decided with account id being null or not.

However in most locations this is a bit cumbersome or hard to follow.

Also: I wasn't able to use this fragment in the ListsActivity (which would make three functions).
Mostly because I don't see how a simple list list could be extended to _also_ support membership info sometimes.
(Would need two wrapped flows for the load state??)

So I would be still open to suggestion how this could be solved more architectually clean and easier to understand.